### PR TITLE
Fix race in StressTestClientTest.gaugesShouldBeExported(). Fixes #1695

### LINKED ISSUE #1695: StressTestClientTest.gaugesShouldBeExported is flaky
As seen at https://travis-ci.org/grpc/grpc-java/jobs/124416631 (since rebuilt):
```
io.grpc.testing.integration.StressTestClientTest > gaugesShouldBeExported FAILED
    java.lang.AssertionError: gauges: [/stress_test/server_0/channel_0/stub_1/qps]
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at io.grpc.testing.integration.StressTestClientTest.gaugesShouldBeExported(StressTestClientTest.java:158)
```

Also seen at https://travis-ci.org/grpc/grpc-java/jobs/124340841 (since rebuilt)

### PR DESCRIPTION
**Pull request by buchgr**
Fixes #1695

### CODE REVIEW

File: interop-testing/src/test/java/io/grpc/testing/integration/StressTestClientTest.java [hunk 1]
```diff
@@ -31,6 +31,7 @@
 
 package io.grpc.testing.integration;
 
+import static com.google.api.client.util.Lists.newArrayList;
```
Comment (carl-mastrangelo) [inline interop-testing/src/test/java/io/grpc/testing/integration/StressTestClientTest.java]: Nope

File: interop-testing/src/test/java/io/grpc/testing/integration/StressTestClientTest.java [hunk 2]
```diff
@@ -132,27 +133,27 @@ public void gaugesShouldBeExported() throws Exception {
...
-    }
-
     Set<String> gaugeNames = newHashSet("/stress_test/server_0/channel_0/stub_0/qps",
         "/stress_test/server_0/channel_0/stub_1/qps");
 
-    while (responseIt.hasNext()) {
-      GaugeResponse response = responseIt.next();
-      String gaugeName = response.getName();
+    List<GaugeResponse> allGauges =
+        newArrayList(stub.getAllGauges(EmptyMessage.getDefaultInstance()));
```
Comment (carl-mastrangelo) [inline interop-testing/src/test/java/io/grpc/testing/integration/StressTestClientTest.java]: Google is moving away from newArrayList.  If you want a copy, use ImmutableList.copyOf().

File: interop-testing/src/test/java/io/grpc/testing/integration/StressTestClientTest.java (hunk 1 above)
Comment (buchgr) [inline interop-testing/src/test/java/io/grpc/testing/integration/StressTestClientTest.java]: ?

Comment (buchgr) [inline interop-testing/src/test/java/io/grpc/testing/integration/StressTestClientTest.java]: oh damn ... again 😅

**Comment by buchgr** [comment]
PTAL

**Comment by carl-mastrangelo** [comment]
@buchgr LGTM

**Comment by buchgr** [comment]
sorry about that stupid import :\

**Comment by buchgr** [comment]
@carl-mastrangelo have u been on the guava team before?

**Comment by carl-mastrangelo** [comment]
No, but they regularly announce best practices internally.

**Comment by carl-mastrangelo** [comment]
Travis is being dumb, merging anyways.

**Pull request merged**